pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -0,0 +1,831 @@
+        }
+      }
+      if (tableConfig.isDimTable()) {
+        Preconditions.checkState(ingestionConfig.getBatchIngestionConfig() != null,
+            "Dimension tables must have batch ingestion configuration");
+      }
+
+      // Stream
+      if (ingestionConfig.getStreamIngestionConfig() != null) {
+        IndexingConfig indexingConfig = tableConfig.getIndexingConfig();
+        Preconditions.checkState(indexingConfig == null || MapUtils.isEmpty(indexingConfig.getStreamConfigs()),
+            "Should not use indexingConfig#getStreamConfigs if ingestionConfig#StreamIngestionConfig is provided");
+        List<Map<String, String>> streamConfigMaps = ingestionConfig.getStreamIngestionConfig().getStreamConfigMaps();
+        Preconditions.checkState(streamConfigMaps.size() == 1, "Only 1 stream is supported in REALTIME table");
+        try {
+          // Validate that StreamConfig can be created
+          new StreamConfig(tableNameWithType, streamConfigMaps.get(0));
+        } catch (Exception e) {
+          throw new IllegalStateException("Could not create StreamConfig using the streamConfig map", e);
+        }
+      }
+
+      // Filter config
+      FilterConfig filterConfig = ingestionConfig.getFilterConfig();
+      if (filterConfig != null) {
+        String filterFunction = filterConfig.getFilterFunction();
+        if (filterFunction != null) {
+          try {
+            FunctionEvaluatorFactory.getExpressionEvaluator(filterFunction);
+          } catch (Exception e) {
+            throw new IllegalStateException("Invalid filter function " + filterFunction, e);
+          }
+        }
+      }
+
+      // Transform configs
+      List<TransformConfig> transformConfigs = ingestionConfig.getTransformConfigs();
+      if (transformConfigs != null) {
+        Set<String> transformColumns = new HashSet<>();
+        for (TransformConfig transformConfig : transformConfigs) {
+          String columnName = transformConfig.getColumnName();
+          if (schema != null) {
+            Preconditions.checkState(schema.getFieldSpecFor(columnName) != null,
+                "The destination column '" + columnName + "' of the transform function must be present in the schema");
+          }
+          String transformFunction = transformConfig.getTransformFunction();
+          if (columnName == null || transformFunction == null) {
+            throw new IllegalStateException(
+                "columnName/transformFunction cannot be null in TransformConfig " + transformConfig);
+          }
+          if (!transformColumns.add(columnName)) {
+            throw new IllegalStateException("Duplicate transform config found for column '" + columnName + "'");
+          }
+          FunctionEvaluator expressionEvaluator;
+          try {
+            expressionEvaluator = FunctionEvaluatorFactory.getExpressionEvaluator(transformFunction);
+          } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)